package/network/utils/umbim/files/lib/netifd/proto/mbim.sh

```diff
@@ -150,8 +150,10 @@ _proto_mbim_setup() {
 	proto_init_update "$ifname" 1
 	proto_send_update "$interface"
 
+	[ -z "$dhcp" ] && dhcp=1
+
 	[ "$pdptype" = "ipv4" -o "$pdptype" = "ipv4v6" ] && {
-		if [ -z "$dhcp" -o "$dhcp" = 0 ]; then
+		if [ -n "$ipv4address" ]; then
```

Comment:
This is closer to the intent of my previous patch, looks good to me.
https://github.com/TDT-AG/openwrt/pull/1/commits/cacbab1605de31470da81e46348b3346d0fc054c